lib/internal/process.js

```diff
@@ -145,7 +145,69 @@ function setupKillAndExit() {
       process._exiting = true;
       process.emit('exit', process.exitCode || 0);
     }
+
+    // Flush stdio streams prior to exit.
+    // `flushSync` not present if stream redirected to file in shell.
+    flushSync(process.stdout);
+    flushSync(process.stderr);
+
     process.reallyExit(process.exitCode || 0);
+
+    function flushSync(stream) {
+
+      // Behavior of this function outside of process.exit() is undefined
+      // due to the following factors:
+      // * Stream fd may be blocking after this call.
+      // * In the event of an incomplete flush pending buffered write
+      //   requests may be truncated.
+      // * No return code.
+
+      if (stream._writev)
+        return;
```

Comment:
I assume if there is `_writev` it already flushes the writes, but I have not checked. (Was kzc's code)